src/Microsoft.AspNetCore.Mvc.Razor.Precompilation/Internal/PrecompileRunCommand.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Hosting.Internal;
+using Microsoft.AspNetCore.Mvc.Razor.Compilation;
+using Microsoft.AspNetCore.Mvc.Razor.Internal;
+using Microsoft.AspNetCore.Razor.CodeGenerators;
+using Microsoft.Cci;
+using Microsoft.Cci.MutableCodeModel;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.ObjectPool;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Precompilation.Internal
+{
+    public class PrecompileRunCommand : PrecompileCommandBase
+    {
+        private const string ConfigureMvcMethod = "ConfigureMvc";
+        private const string PrecompiledAssemblyPrefix = "PrecompiledRazor.";
+
+        private IMvcRazorHost Host { get; set; }
+
+        private IRoslynCompilationService CompilationService { get; set; }
+
+        private Action<IMvcBuilder> ConfigureMvcAction { get; set; }
+
+        protected override int ExecuteCore()
+        {
+            ParseArguments();
+
+            var services = ConfigureDefaultServices();
```

Comment:
Abstract these two lines into an MvcRazorHostProvider. Too much going on in this class